### PR TITLE
ENH: add shellcheck run on shell scripts which ATM are shellcheck kosher

### PR DESCRIPTION
**Pull request by yarikoptic**
I find `shellcheck` really handy!

I thought to make this workflow conditioned on changes to those files, but IMHO too much hassle for what it is worth since shellcheck run should be quite fast

### CODE REVIEW

**Comment by yarikoptic** [comment]
grr, it didn't run... so I screwed up somewhere but I don't see where and there is no annotation of any kind from the system: https://github.com/yarikoptic/datalad/actions/runs/279238347   may be someone else could see the gotcha

**Comment by kyleam** [comment]
> grr, it didn't run...

Should be fixed now.

**Comment by yarikoptic** [comment]
woohoo - thank you @kyleam !

File: .github/workflows/shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        sudo apt-get install shellcheck
+    - uses: actions/checkout@v1
+    - name: Run shellcheck
+      run: |
+        shellcheck \
+          tools/bisect-git-annex.scripts/bisect-git-annex-lock.sh \
+          tools/ci/install-annex.sh \
+          tools/ci/install-minimum-git.sh \
+          tools/ci/prep-travis-devel-annex.sh \
+          tools/ci/prep-travis-forssh.sh
```
Comment (kyleam) [inline .github/workflows/shellcheck.yml:24]: > which ATM are shellcheck kosher

I think it's worth adding a code comment explaining that you're only testing the subset of scripts that pass shellcheck.

I was curious about what this looked like overall:

```console
$ git grep -l '^#!/.*sh$' tools/ | wc
     29      29     851
$ shellcheck $(git grep -l '^#!/.*sh$' tools/) | grep -c "In tools/"
144
```

File: .github/workflows/shellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    steps:
+    - name: Set up system
+      shell: bash
+      run: |
+        sudo apt-get update -qq
+        sudo apt-get install shellcheck
+    - uses: actions/checkout@v1
+    - name: Run shellcheck
+      run: |
```
Comment (yarikoptic) [inline .github/workflows/shellcheck.yml:18]:
```suggestion
      run: |
        # I: running only on a subset of scripts which are shellcheck clean ATM
```

File: .github/workflows/shellcheck.yml (hunk 1 above)
Comment (yarikoptic) [reply .github/workflows/shellcheck.yml:25]: added a comment, thanks.  Some scripts just need minor fixups to become shellcheck compliant, but I did not embark on that mission

**Pull request merged**